### PR TITLE
Add --shallow option in wstool init.

### LINKED ISSUE #96: Speedup CI by doing shallow clones
wstool seems to lack the ability to only download the latest code in a git repo, e.g.:

```
git clone --depth 1 https://github.com/jquery/jquery.git jquery
```

Seems like this should be a command-line option for `wstool update`. This should greatly improve download time for many repos with long histories.

### PR DESCRIPTION
**Pull request by at-wat**
This commit fixes #96.

### CODE REVIEW

**Comment by tkruse** [comment]
Would be good to have at least one integration test using shallow=true for a git repo

**Comment by at-wat** [comment]
@tkruse Tests to check behavior of `--shallow` option has been added. PTAL

**Comment by tkruse** [comment]
lgtm, +1

File: test/local/test_shallow_checkout_git.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2009, Willow Garage, Inc.
+# Copyright (c) 2017, wstool authors
```
Comment (130s) [inline test/local/test_shallow_checkout_git.py]: :sweat_smile:

**Pull request merged**